ema_workbench/em_framework/optimization.py

```diff
@@ -546,6 +745,32 @@ def __call__(self, optimizer):
             pass
 
 
+def epsilon_nondominated(results, epsilons, problem):
+    """Merge the list of results into a single set of
+    non dominated results using the provided epsilon values
+    Parameters
+    ----------
+    results : list of DataFrames
+    epsilons : epsilon values for each objective
+    problem : PlatypusProblem instance
+    Returns
+    -------
+    DataFrame
+    Notes
+    -----
+    this is a platypus based alternative to pareto.py (https://github.com/matthewjwoodruff/pareto.py)
+    """
+    if problem.nobjs != len(epsilons):
+        ValueError("the number of epsilon values must match the number of objectives")
```

Comment:
Also give the numbers back to help with debugging.

```suggestion
        ValueError(f"the number of epsilon values ({len(epsilons)}) must match the number of objectives {problem.nobjs}")
```

Note that this could also be done  walrus operator (`:=`) could here also be used twice, but I found it a bit overkill. Using it, it would look like:

```python
    if o := problem.nobjs != e := len(epsilons):
        ValueError(f"the number of epsilon values ({e}) must match the number of objectives {o}")
```
But it read a lot worse, in my opinion (it's faster though, if you ever need performance in such a construction).